### PR TITLE
Support x-forwarded-for to work behind proxies

### PR DESCRIPTION
**Pull request by jrudess**
x-forwarded-for can be a list, so using x-real-ip

### CODE REVIEW

File: defaults.js [hunk 1]
```diff
@@ -41,6 +41,9 @@ module.exports = {
   // Path to keep the moderation log
   modlog: "mod.log",
 
+  // Allow X-Forwarded-For to set IP if found
+  useXForwardedFor: false,
```
Comment (RealDolos) [inline defaults.js]: Don't really like the name... how about `considerProxyForwardedForHeaders`?

File: lib/client.js [hunk 2]
```diff
@@ -73,7 +73,8 @@ class Client {
   constructor(room, socket, rtoken) {
     const {remoteAddress, remotePort} = socket.request.connection;
     this.socket = socket;
-    this.ip = socket.request.headers['x-real-ip'] || remoteAddress;
+    let xForwardFor = CONFIG.get("useXForwardedFor") ? socket.request.headers["x-forwarded-for"] : null;
+    this.ip = xForwardFor ? xForwardFor.split(",")[0] : remoteAddress;
```
Comment (RealDolos) [inline lib/client.js]: two tertiaries and a null gibs headaches :*/
please for the love of god, don't. There is nothing wrong with an if or two...

Comment (jrudess) [reply lib/client.js]: Wuuuuut.   Trinary is akin to the holy trinity.

File: lib/client.js [hunk 3]
```diff
@@ -73,7 +73,12 @@ class Client {
   constructor(room, socket, rtoken) {
     const {remoteAddress, remotePort} = socket.request.connection;
     this.socket = socket;
-    this.ip = remoteAddress;
+    if (CONFIG.get("considerProxyForwardedForHeaders")) {
+      this.ip = socket.request.headers["x-forwarded-for"] || remoteAddress;
+      this.ip = this.ip.split(",")[0];
+    } else {
```
Comment (RealDolos) [inline lib/client.js]: >else on same line
kys

Comment (jrudess) [reply lib/client.js]: The one-true-brace-style will someday win.  You will see.

**Pull request merged**